include/oneapi/mkl/detail/backend_selector.hpp

```diff
@@ -27,6 +27,8 @@
 namespace oneapi {
 namespace mkl {
 
+using namespace cl;
```

Comment:
In case the `CL/sycl.hpp` header is included only the `cl::sycl::` namespace is present in hipSYCL. Changing the includes to include `sycl/sycl.hpp` does not seem to affect dpc++, and that way we wouldn't need to use the `cl::` namespace.

Reply:
@mkrainiuk The `CL/sycl.hpp` header is deprecated in SYCL 2020 and only there for backwards compatibility. The SYCL specification as it is currently worded only says:

> For compatibility with SYCL 1.2.1, SYCL provides another standard header file: <CL/sycl.hpp>, which can be included in place of <sycl/sycl.hpp>. In that case, all SYCL classes, constants, types and functions defined by this specification should exist within the ::cl::sycl C++ namespace.
> For consistency, the programming API will only refer to the <sycl/sycl.hpp> header and the ::sycl namespace, but this should be considered synonymous with the SYCL 1.2.1 header and namespace.

The problem is that when including `CL/sycl.hpp`, DPC++ exposes `::sycl` as well, which is arguably incorrect, although the specification is not super clear here. hipSYCL follows a more literal interpretation of this section of the specification where `CL/sycl.hpp` exposes `::cl::sycl` and `sycl/sycl.hpp` exposes `::sycl`.

See here for a discussion of exactly this issue: https://github.com/KhronosGroup/SYCL-CTS/pull/108

I agree that the proper solution is to move to the SYCL 2020 header `sycl/sycl.hpp` and use `::sycl`, but I'm not sure how well this is already supported in DPC++.

Reply:
Thanks @sbalint98 and @illuhad! Sorry for delay. I agree that we need to move to SYCL 2020, I also checked sycl/sycl.hpp in DPC++ compiler, and as I see it's not fully supported yet, so I'm fine with having cl namespace, we can change this part when both hipSYCL and DPC++ compilers will fully move to SYCL 2020.